src/sprites/sprite.js

```diff
@@ -140,11 +149,41 @@ class Sprite {
         const blocksContainer = this.blocks._blocks;
         const originalBlocks = Object.keys(blocksContainer).map(key => blocksContainer[key]);
         const copiedBlocks = JSON.parse(JSON.stringify(originalBlocks));
-        newBlockIds(copiedBlocks);
+        const oldToNew = newBlockIds(copiedBlocks);
         copiedBlocks.forEach(block => {
             newSprite.blocks.createBlock(block);
         });
 
+        Object.keys(this.comments).forEach(commentId => {
+            const comment = this.comments[commentId];
+            const newComment = new Comment(
+                null, // use new comment id
+                comment.text,
+                comment.x,
+                comment.y,
+                comment.width,
+                comment.height,
+                comment.minimized
+            );
+            if (comment.blockId) {
```

Comment:
```suggestion
            // If the comment is attached to a block, it has a blockId property for the block it's attached to.
            // Because we generate new block IDs for all the duplicated blocks, change all the comments' attached-block
            // IDs from the old block IDs to the new ones.
            if (comment.blockId) {
```